bucketlist/app.py

```diff
@@ -0,0 +1,281 @@
+            else:
+                return({'message': 'wrong username/password combination'}, 401)
+
+    def get(self):
+        '''Ensures that get method not allowed in logging in'''
+        return({'message': 'Method is not allowed'}, 405)
+
+
+class BucketLists(Resource):
+    '''Adds and retrieves bucketlists to and from the database'''
+    @auth.login_required
+    def get(self):
+        # Adds page and limit variables to be used for pagination
+        page = request.args.get('page', 1)
+        limit = request.args.get('limit', 1) or 100
+        # Adds a query attribute specified in the url
+        q = request.args.get('q')
+        if q:
+            # Searches bucketlists for one with the given query
+            buckets = Bucketlist.query.filter(Bucketlist.bucketlist_name.ilike(
+                '%' + q + "%")).filter_by(created_by=g.user.id).paginate(int(page), int(limit), False)
+            bkts = buckets.items
+            if buckets:
+                if buckets.has_next:
+                    next_page = str(
+                        request.url_root) + 'bucketlist/api/bucketlists?q=' + q + '&page=' + str(int(page) + 1)
+                else:
+                    next_page = 'None'
+
+                if buckets.has_prev:
+                    prev = str(request.url_root) + 'bucketlist/api/bucketlists?q=' + \
+                        q + '&page=' + str(int(page) - 1)
+                else:
+                    prev = 'None'
+                buckets = [bucket for bucket in bkts]
+                response = {'bucketlists': marshal(
+                    buckets, Bucketlist_marshaller),
+                    'next': next_page,
+                    'prev': prev}
+                return response
+            else:
+                return {'message': ' Bucketlist not found'}, 404
+
+        else:
+            buckets = Bucketlist.query.filter_by(
+                created_by=g.user.id).paginate(int(page), int(limit), False)
+            bkts = buckets.items
+            buckets = [bucket for bucket in bkts]
+            return marshal(buckets, Bucketlist_marshaller)
+            if buckets:
+                if buckets.has_next:
+                    next_page = str(
+                        request.url_root) + \
+                        'bucketlist/api/bucketlists?limit=' + str(int(limit)) + \
+                        '&page=' + str(int(page) + 1)
+                else:
+                    next_page = 'None'
+
+                if buckets.has_prev:
+                    prev = str(request.url_root) + 'bucketlist/api/bucketlists?limit=' + \
+                        str(int(limit)) + '&page=' + str(int(page) - 1)
+
+                else:
+                    prev = 'None'
+                buckets = [bucket for bucket in bkts]
+                response = {'bucketlists': marshal(
+                    buckets, Bucketlist_marshaller),
+                    'next': next_page,
+                    'prev': prev}
+                return response
```

Comment:
I still think their is a better way of writing the line 0f code 93-150. There is a lot of repetition.